### DISCUSSION TITLE
Design

### DISCUSSION THREAD

**Discussion by Luciffffer** [OSTA-group/osta#82] [General]
Is there already some sort of design you guys base the app on?
Like a figma file or something like that?

**Comment by BauwenDR** [comment]
We used a tool called Miro to create our initial wireframes ([wireframes](https://miro.com/app/board/uXjVKUO1Wrw=/)).

In terms of custom components, we did not do a lot of styling ourselves as we used the default Ionic components. We only really swapped the primary color to more closely resemble our app logo.

In case you are looking for a figma file (or similar) I would suggest starting with an Ionic template.

**Comment by Luciffffer** [comment]
Nice the wireframes look great!

I have some ideas for a more advanced design that I'd like to share.
Just some inspiration in case anyone ever wants to build something along those lines. Maybe I'll give it a shot at it when I have more time.

Here's the figma link where I'm currently working it out a little:
[Figma design link](https://www.figma.com/design/c3ZVKkE4evw6PITO6N75Q7/OSTA-Design?node-id=0-1&t=uCZQTn0iheSITPJZ-1)

**Comment by BauwenDR** [comment]
The design looks really interesting. I like the increased easy of showing landmark details. This could be a fix for the popup markers not working (as they can be replaced by the slider). Oh and the loading screen is pretty cool.

There are some additional things to consider:
- Nearby landmarks, we can only know what landmarks are the closest after calculating the distances between you and all loaded landmarks, optimisations might be needed to filter landmarks for area as sorting 1000 landmarks each time will be computationally heavy (especially if we recalculate every time a user moves). Generally I do not yet know how feasible this is, but is definitely a nice to have.
- I created a discussion on having more ways of creating/ editing a trip. If you don't mind you can have a look at it and try to see if you can incorporate some of my ideas (https://github.com/OSTA-group/osta/discussions/84).

That being said, I am looking into releasing v1.1.0 sooner rather than later, so the bigger changes might be postponed until version 1.2.0. I have created an issue for the loading screen and intend to finish it for 1.1.0.
